src/core/replicators.ts

```diff
@@ -0,0 +1,190 @@
+import * as tt from '../../typings/telegram-types'
+import Markup from '../markup'
+const { formatHTML } = Markup
+
+type Replicate<X, K extends keyof (tt.Message & X)> = Required<
+  Pick<tt.Message & X, K>
+>
+type Copy<M extends keyof tt.Telegram> = tt.MakeExtra<M, never>
```

Comment:
`never` could be the default value for 2nd type param of `tt.MakeExtra`, allowing it to be used directly.